core/tracing/hooks.go

```diff
@@ -0,0 +1,316 @@
+	OnGenesisBlock    GenesisBlockHook
+	OnSystemCallStart OnSystemCallStartHook
+	OnSystemCallEnd   OnSystemCallEndHook
+	// State events
+	OnBalanceChange BalanceChangeHook
+	OnNonceChange   NonceChangeHook
+	OnCodeChange    CodeChangeHook
+	OnStorageChange StorageChangeHook
+	OnLog           LogHook
+}
+
+// BalanceChangeReason is used to indicate the reason for a balance change, useful
+// for tracing and reporting.
+type BalanceChangeReason byte
+
+//go:generate go run golang.org/x/tools/cmd/stringer -type=BalanceChangeReason -output gen_balance_change_reason_stringer.go
+
+const (
+	BalanceChangeUnspecified BalanceChangeReason = 0
+
+	// Issuance
+	// BalanceIncreaseRewardMineUncle is a reward for mining an uncle block.
+	BalanceIncreaseRewardMineUncle BalanceChangeReason = 1
+	// BalanceIncreaseRewardMineBlock is a reward for mining a block.
+	BalanceIncreaseRewardMineBlock BalanceChangeReason = 2
+	// BalanceIncreaseWithdrawal is ether withdrawn from the beacon chain.
+	BalanceIncreaseWithdrawal BalanceChangeReason = 3
+	// BalanceIncreaseGenesisBalance is ether allocated at the genesis block.
+	BalanceIncreaseGenesisBalance BalanceChangeReason = 4
+
+	// Transaction fees
+	// BalanceIncreaseRewardTransactionFee is the transaction tip increasing block builder's balance.
+	BalanceIncreaseRewardTransactionFee BalanceChangeReason = 5
+	// BalanceDecreaseGasBuy is spent to purchase gas for execution a transaction.
+	// Part of this gas will be burnt as per EIP-1559 rules.
+	BalanceDecreaseGasBuy BalanceChangeReason = 6
+	// BalanceIncreaseGasReturn is ether returned for unused gas at the end of execution.
+	BalanceIncreaseGasReturn BalanceChangeReason = 7
+
+	// DAO fork
+	// BalanceIncreaseDaoContract is ether sent to the DAO refund contract.
+	BalanceIncreaseDaoContract BalanceChangeReason = 8
+	// BalanceDecreaseDaoAccount is ether taken from a DAO account to be moved to the refund contract.
+	BalanceDecreaseDaoAccount BalanceChangeReason = 9
+
+	// BalanceChangeTransfer is ether transferred via a call.
+	// it is a decrease for the sender and an increase for the recipient.
+	BalanceChangeTransfer BalanceChangeReason = 10
+	// BalanceChangeTouchAccount is a transfer of zero value. It is only there to
+	// touch-create an account.
+	BalanceChangeTouchAccount BalanceChangeReason = 11
+
+	// BalanceIncreaseSelfdestruct is added to the recipient as indicated by a selfdestructing account.
+	BalanceIncreaseSelfdestruct BalanceChangeReason = 12
+	// BalanceDecreaseSelfdestruct is deducted from a contract due to self-destruct.
+	BalanceDecreaseSelfdestruct BalanceChangeReason = 13
+	// BalanceDecreaseSelfdestructBurn is ether that is sent to an already self-destructed
+	// account within the same tx (captured at end of tx).
+	// Note it doesn't account for a self-destruct which appoints itself as recipient.
+	BalanceDecreaseSelfdestructBurn BalanceChangeReason = 14
+
+	// OP-Geth specific
+	BalanceMint BalanceChangeReason = 200
+)
```

Comment:
Added this constant for tracing balance mints (see `state_transition.go` for where it's used).

Reply:
I updated the live supply-tracer to capture this trace type balance change